### PR TITLE
Update GHC and package dependency version pinning

### PR DESCRIPTION
**Pull request by atomb**

### CODE REVIEW

File: .github/workflows/build.yml
Comment (RyanGlScott) [inline .github/workflows/build.yml:69]: Based off of what @jared-w has told me, `actions/setup-haskell` is out of date and `haskell/actions/setup` should be used instead. I'm not sure about the precise version number, but I believe `@v1` should suffice.

Comment (atomb) [reply .github/workflows/build.yml:69]: I just noticed his PR that does essentially the same thing as this one. :) I guess we should only merge one of them!

Comment (atomb) [reply .github/workflows/build.yml]: Updated

**Comment by RyanGlScott** [comment]
As far as why this is failing on Windows in particular, see http://hackage.haskell.org/package/hsc2hs-0.68.7/hsc2hs.cabal. To work around a Windows-specific `process` bug, that version of `hsc2hs`requires `process >= 1.6.8`, which is more recent that what is pinned in the `.config` file.

One possible way to fix this would be to re-`freeze` the files but with `hsc2hs==0.68.6`, which doesn't have quite as restrictive version bounds on `process` for Windows.

**Comment by RyanGlScott** [comment]
This looks sensible, although I worry that we'll have similar difficulties later when we try to update the `.config` files. Can you document somewhere the series of additional steps you needed to apply on top of `cabal freeze` to make things work? Better yet would be scripting these additional scripts, but perhaps that's not feasible.

**Comment by atomb** [comment]
I ultimately want this to be completely automated, and think it could be, but it'll take more work, and until we do at least part of this commit we can't build using GHC 8.10.3, which it's important for us to be able to do.

What I'd like in the long run would be to have some sort of automated process that would try running `cabal freeze` on each platform, try to build, and create a PR if the build succeeds. That way we could keep building against the latest libraries, but only through an easy, opt-in process.

There were some manual changes needed here because of some reorganization we've been doing. After this merge, I think the only manual changes would be to remove platform-specific entries. Alternatively, we could have separate freeze files for Windows (which we've done in the past), which should eliminate any need to do manual work.

**Comment by RyanGlScott** [comment]
OK. I'm fine with merging this as-is, but I do think it's worth writing down what the platform-specific packages are somewhere, as that could save someone else some pain down the road.

**Comment by atomb** [comment]
Yes, that's a very good point. I just pushed a commit to the README outlining what packages to leave out of freeze files.

**Review by RyanGlScott — APPROVED** [review]
Looks good to me.

Now that we have [Mergify enabled](https://github.com/GaloisInc/saw-script/pull/1053/files), I'm going to try to see if I can get this automatically merged once CI passes by applying the new "ready-to-merge" label.

**Comment by atomb** [comment]
Whee! That'll be awesome. :) I'm so happy to have Mergify.

**Comment by brianhuffman** [comment]
@atomb If you want to give your approval to #1059 then we can see have a race and see how mergify handles it

**Comment by brianhuffman** [comment]
@atomb: So #1059 got merged first, and now this one's out of date with master. I think we need a "keep-updated" label if we want Mergify to automatically rebase and restart the CI when that happens. (That label doesn't exist yet; we should double check to make sure we name it the right thing.)

Also, I see some CI failures with the message "Error: retrieving gpg key timed out" which probably have nothing to do with this PR.

**Pull request merged**
